src/transaction/log_append.cpp

```diff
@@ -184,6 +184,7 @@ LOG_RESET_APPEND_LSA (const LOG_LSA *lsa)
   // todo - concurrency safe-guard
   log_Gl.hdr.append_lsa = *lsa;
   log_Gl.prior_info.prior_lsa = *lsa;
+  log_Gl.m_prior_sender.reset_unsent_lsa (*lsa);
```

Comment:
Don't think the function `reset_unsent_lsa` is needed.
Is it that the combination of `reset_unsent_lsa` and the checks (assert) inside `prior_sender::send_list` is intended to ensure than no log prior lists are "lost" between the initialization moment (`LOG_RESET_APPEND_LSA`) and the actual first call to `send_list`?

Because I think that this can be solved in a different manner: enforce that `send_list` is not called without having sinks hooks registered.

I debugged and the chronology of the ATS boot-up is somethink like this:
- `prior_sender` ctor is being called
- connection to all PSs in the static configuration is done
  - the log prior sink hooks to dispatch log prior lists to PS are registered
- the ATS choses the "main connection" (for now, the first one active)
  - but in the future a "leader PS" will be elected
- `tran_server::get_boot_info_from_page_server`
- ATS requests the log header page from main PS - `logpb_fetch_header_from_page_server`
- ATS requests last log page - I think - from main PS - another call to `logpb_fetch_header_from_page_server`
- `LOG_RESET_APPEND_LSA` is called

Reply:
> hasn't been called yet, but a PS requests the catch-up LSA not on booting.

Isn't this a practically impossible situation?

Reply:
It seems allowed by design, and I presume it's possible. If you start cub_server after a normal shutdown with develop, it doesn't do recovery and only `LOG_DUMMY_HA_SERVER_STATE` is appended, which I don't think is necessary for a database.